### DISCUSSION TITLE
Change the double control-c thing

### DISCUSSION THREAD

**Discussion by dmeleedy** [ggml-org/llama.cpp#1523] [Ideas]
I often get large amounts of text that I want to control-c on, and then the prompt shows up and I lose my entire session.   Or I hit control-c twice by mistake because there is no indication on the screen that I hit the first control-c. Can you make interruptions to the chat one control character, and the interruption to stop the entire program another control character please?  This has been driving me completely crazy, as I always lose a ton of context while writing stories.

**Comment by hungerf3** [comment]
I just remove that part of the code so that control-c always just sets the interacting flag, and use control-shift-| to send a sigquit to exit. 

Just edit the sigint_handler function in examples/main/main.cpp and remove the _exit(130);

**Reply by dmeleedy** [reply]
you wouldn't happen to have a handy dandy diff of that would you?

**Reply by dmeleedy** [reply]
Nevermind, I was able to do it easily once I found it.. Thank you so much for this tip
